pkg/compactor/compactor.go

```diff
@@ -792,10 +792,20 @@ func (c *Compactor) compactUserWithRetries(ctx context.Context, userID string) e
 		if lastErr == nil {
 			return nil
 		}
+		if ctx.Err() != nil {
+			level.Warn(util_log.WithContext(ctx, c.logger)).Log("msg", "received context error during compaction", "org_id", "err", ctx.Err())
+			return ctx.Err()
+		}
 		if c.isCausedByPermissionDenied(lastErr) {
-			level.Warn(c.logger).Log("msg", "skipping compactUser due to PermissionDenied", "user", userID, "err", lastErr)
+			level.Warn(c.logger).Log("msg", "skipping compactUser due to PermissionDenied", "org_id", userID, "err", lastErr)
```

Comment:
Is there a specific reason to change this to `org_id`? I think we use `user` here everywhere in compactor

Reply:
We use `org_id` everywhere: https://github.com/cortexproject/cortex/blob/master/pkg/util/log/wrappers.go#L18

Reply:
I believe we have quite a lot of logs using user as key in compactor. Do we plan to change all of them

Reply:
I reverted the log change for now. Would address it in separate PR. Would utilize the log wrapper where possible in compactor